Desktop/analysis/analysis.cpp

```diff
@@ -835,6 +864,118 @@ void Analysis::setErrorInResults(const std::string & msg)
 	setResults(errorResults, Status::FatalError);
 }
 
+void Analysis::setFormParent(QQuickItem *formParent)
+{
+	if(_formParent == formParent)
+		return;
+
+	//Log::log() << "Analysis::setFormParent(" << formParent << ") " << name() << std::endl;
+
+	_formParent = formParent;
+
+	emit formParentChanged();
+
+	if(_formParent)
+	{
+		if(_analysisForm)				_analysisForm->setParentItem(_formParent);
+		else if(readyToCreateForm())	createForm();
+	}
+}
+
+bool Analysis::readyToCreateForm() const
+{
+	return _formParent && dynamicModule() && dynamicModule()->readyForUse();
+}
+
+void Analysis::createForm()
+{
+	Log::log() << "Analysis(" << this << ")::createForm() called ";
+
+	setQmlError("");
+
+	if(!_formParent)
+	{
+		Log::log() << "There is no formParent..." << std::endl;
+		return;
+	}
+
+	if(_analysisForm)
+	{
+		Log::log() << "It already has a form, so we destroy it.";
+		destroyForm();
+	}
+
+	try
+	{
+		Log::log()  << std::endl << "Loading QML form from: " << qmlFormPath(false, false) << std::endl;
+
+		QObject * newForm = instantiateQml(QUrl::fromLocalFile(tq(qmlFormPath(false, false))), module(), qmlContext(_formParent));
+
+		Log::log() << "Created a form, got pointer " << newForm << std::endl;
+
+		AnalysisForm * formItem = qobject_cast<AnalysisForm *>(newForm);
+
+		if(!formItem)
+			throw Modules::ModuleException(module(), "QML file '" + qmlFormPath(false, false) + "' didn't spawn into AnalysisForm, but into: " + (newForm ? fq(newForm->objectName()) : "null"));
+
+		setAnalysisForm(formItem);
+
+		formConnect();
+	}
+	catch(Modules::ModuleException me)
+	{
+		Log::log() << "Had an error loading qml: " << me.what() << std::endl;
+		setQmlError(me.what());
+		_analysisForm = nullptr;
+	}
+	catch(std::exception e)
+	{
+		setQmlError(e.what());
+		_analysisForm = nullptr;
+	}
+}
+
+void Analysis::destroyForm()
+{
+	Log::log() << "Analysis(" << this << ")::destroyForm() called";
+
+	if(_analysisForm)
+	{
+		Log::log(false) << " it has a AnalysisForm " << _analysisForm << " so let's destroy it" << std::endl;
+
+		_optionsDotJASP = _boundValues;  //So that we can later reload the controls to what we currently have
+
+		_analysisForm->setParent(		nullptr);
+		_analysisForm->setParentItem(	nullptr);
+
+		delete _analysisForm;
+		_analysisForm = nullptr;
```

Comment:
@boutinb `destroyForm()` always clears `_analysisForm`.

Reply:
Yes, but if the parent of the form is destroyed before, QML will destroy the form without going to this destroyForm method, making _analysisForm a dangling pointer.

Reply:
As we've just talked about, it will be fine